llvm/test/CodeGen/AIE/aie2/ra/waw_reg_renaming_loop.mir

```diff
@@ -0,0 +1,609 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+# NOTE: Example file for Write After Write Register Renaming in Loop test
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+# unit test for the WAW register renaming pass and check edge cases
+# RUN: llc -mtriple=aie2 -verify-machineinstrs --start-before=greedy --stop-after=virtregrewriter %s -o - | FileCheck %s
+#
+# disable pass to highlight the changes the WAW register rewriting pass performs
+# RUN: llc -mtriple=aie2 -verify-machineinstrs --start-before=greedy --stop-after=virtregrewriter --aie-wawreg-rewrite=false %s -o - | FileCheck %s --check-prefix=DISABLE-PASS
```

Comment:
Nit: I would really just have the first RUN line. Having this one really just tests greedy, and makes the test cases a bit too bloated IMO. What do you think @andcarminati @martien-de-jong ?

Reply:
simplified the test. The comments should help explain what the individual unit tests are actually doing